internal/identity/identity.go

```diff
@@ -0,0 +1,199 @@
+package identity
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+
+	"github.com/alecthomas/types/optional"
+	"google.golang.org/protobuf/proto"
+
+	v1 "github.com/TBD54566975/ftl/backend/protos/xyz/block/ftl/v1"
+	"github.com/TBD54566975/ftl/internal/model"
+)
+
+type Identity interface {
+	String() string
+}
+
+var _ Identity = Controller{}
+
+type Controller struct{}
+
+func NewController() Controller {
+	return Controller{}
+}
+
+func (c Controller) String() string {
+	return "ca"
+}
+
+var _ Identity = Runner{}
+
+// Runner identity
+// TODO: Maybe use KeyType[T any, TP keyPayloadConstraint[T]]?
+type Runner struct {
+	Key    model.RunnerKey
+	Module string
+}
+
+func NewRunner(key model.RunnerKey, module string) Runner {
+	return Runner{
+		Key:    key,
+		Module: module,
+	}
+}
+
+func (r Runner) String() string {
+	return fmt.Sprintf("%s:%s", r.Key, r.Module)
+}
+
+func Parse(s string) (Identity, error) {
+	if s == "" {
+		return nil, fmt.Errorf("empty identity")
+	}
+	parts := strings.Split(s, ":")
+
+	if parts[0] == "ca" {
+		return Controller{}, nil
+	}
+
+	key, err := model.ParseRunnerKey(parts[0])
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse key: %w", err)
+	}
+
+	return NewRunner(key, parts[1]), nil
+}
+
+// Store is held by a node and contains the node's identity, key pair, signer, and certificate.
+type Store struct {
+	Identity           Identity
+	KeyPair            KeyPair
+	Signer             Signer
+	Certificate        optional.Option[Certificate]
+	ControllerVerifier optional.Option[Verifier]
+}
+
+func NewStoreNewKeys(identity Identity) (*Store, error) {
+	pair, err := GenerateKeyPair()
+	if err != nil {
+		return nil, fmt.Errorf("failed to generate key pair: %w", err)
+	}
+
+	signer, err := pair.Signer()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get signer: %w", err)
+	}
+
+	return &Store{
+		Identity: identity,
+		KeyPair:  pair,
+		Signer:   signer,
+	}, nil
+}
+
+func (s *Store) NewGetCertificateRequest() (v1.GetCertificationRequest, error) {
```

Comment:
Why Certificate vs. Certification here?

Reply:
Trying to be fancy, I will make it more consistent!